File: blind75/01_two_sum/two_sum.go
package two_sum

// TwoSum returns indices of two numbers that add up to target.
// Uses a hash map for O(n) time complexity.
func TwoSum(nums []int, target int) []int {
	seen := make(map[int]int)

	for i, num := range nums {
		complement := target - num
		if j, found := seen[complement]; found {
			return []int{j, i}
		}
		seen[num] = i
	}

	return nil
}

File: blind75/01_two_sum/two_sum.py
from typing import List


def two_sum(nums: List[int], target: int) -> List[int]:
    """
    Returns indices of two numbers that add up to target.
    Uses a hash map for O(n) time complexity.

    Args:
        nums: List of integers
        target: Target sum

    Returns:
        List containing indices of the two numbers that add up to target
    """
    seen = {}

    for i, num in enumerate(nums):
        complement = target - num
        if complement in seen:
            return [seen[complement], i]
        seen[num] = i

    return []
